File: leetcode/0059/generateMatrix.swift
func generateMatrix(_ n: Int) -> [[Int]] {
    var matrix = Array.init(repeating: Array.init(repeating: 0, count: n), count: n)
    var left = 0, right = n - 1, top = 0, bottom = n - 1
    var insertN = 1
    while left <= right && top <= bottom {
        for column in left..<right+1 {
            matrix[top][column] = insertN
            insertN += 1
        }
        for row in top+1..<bottom+1 {
            matrix[row][right] = insertN
            insertN += 1
        }
        if left < right && top < bottom {
            for column in stride(from: right-1, to: left, by: -1) {
                matrix[bottom][column] = insertN
                insertN += 1
            }
            for row in stride(from: bottom, to: top, by: -1) {
                matrix[row][left] = insertN
                insertN += 1
            }
        }
        left = left + 1
        right = right - 1
        top = top + 1
        bottom = bottom - 1
    }
    
    return matrix
}
print(generateMatrix(3))

File: leetcode/0059/generateMatrix.py
def generateMatrix(n: int) -> [[int]]:
	matrix = [[0 for _ in range(n)] for _ in range(n)]

	left, right, top, bottom = 0, n - 1, 0, n - 1
	insert_n = 1
	while left <= right and top <= bottom:
		for column in range(left, right+1):
			matrix[top][column] = insert_n
			insert_n += 1
		for row in range(top+1, bottom+1):
			matrix[row][right] = insert_n
			insert_n += 1
		if left < right and top < bottom:
			for column in range(right - 1, left, -1):
				matrix[bottom][column] = insert_n
				insert_n += 1
			for row in range(bottom, top, -1):
				matrix[row][left] = insert_n
				insert_n += 1
		left, right, top, bottom = left + 1, right - 1, top + 1, bottom - 1
	return matrix

print(generateMatrix(3))